service/src/main/java/bio/terra/pipelines/common/utils/FileUtils.java

```diff
@@ -25,6 +31,41 @@ public static String getBlobNameFromTerraWorkspaceStorageHttpUrl(
         blobHttpUrl.indexOf(workspaceId.toString()) + workspaceId.toString().length() + 1);
   }
 
+  /**
+   * Construct the destination blob name for a user-provided file input.
+   *
+   * <p>For example, file `local/path/to/file.txt` for jobId `1234` returns
+   * `user-input-files/1234/file.txt`
+   *
+   * @param jobId
+   * @param userProvidedFileInputValue
+   * @return blobName
+   */
+  public static String constructBlobNameForUserInputFile(
```

Comment:
would rename to `constructDestinationBlobNameForUserInputFile` based on your description of it.